python/hail/methods/misc.py

```diff
@@ -403,5 +390,4 @@ def window_by_locus(mt: MatrixTable, bp_window_size: int) -> MatrixTable:
     -------
     :class:`.MatrixTable`
     """
-    require_partition_key_locus(mt, 'window_by_locus')
     return MatrixTable(mt._jvds.windowVariants(bp_window_size))
```

Comment:
This needs to be replaced by some requirement pertaining to locus being the key.

Reply:
I added `require_first_key_field_locus` to replace `require_partition_key_locus`.